Sources/SnappingLayout/Classes/SnappingLayout.swift

```diff
@@ -29,52 +31,120 @@ public class SnappingLayout: UICollectionViewFlowLayout {
         guard let collectionView = collectionView else {
             return super.targetContentOffset(forProposedContentOffset: proposedContentOffset, withScrollingVelocity: velocity)
         }
-
         var offsetAdjusment = CGFloat.greatestFiniteMagnitude
         let horizontalPosition: CGFloat
-
-        switch snapPosition {
-        case .left:
-            horizontalPosition = proposedContentOffset.x + collectionView.contentInset.left + sectionInset.left
-        case .center:
-            horizontalPosition = proposedContentOffset.x + (collectionView.bounds.width * 0.5)
-        case .right:
-            horizontalPosition = proposedContentOffset.x + collectionView.bounds.width - sectionInset.right
-        }
-
-        let targetRect = CGRect(x: proposedContentOffset.x, y: 0, width: collectionView.bounds.size.width, height: collectionView.bounds.size.height)
-        let layoutAttributesArray = super.layoutAttributesForElements(in: targetRect)
-        layoutAttributesArray?.forEach { layoutAttributes in
-            let itemHorizontalPosition: CGFloat
-
+        let verticalPosition: CGFloat
+        
+        switch scrollDirection {
+        case .horizontal:
```

Comment:
Instead of chaining 2 `switch` statements inside another `switch`, you can do it like this:
```suggestion
        switch (scrollDirection, snapPosition) {
        case (.horizontal, .left):
        case (.horizontal, .center):
        case (.horizontal, .right):
        case (.vertical, .center):
        case (.vertical, .top):
        case (.vertical, .bottom):
        case (.horizontal, .top), (.horizontal, .bottom),  (.vertical, .left),  (.vertical, .right):
```